stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/channel/add/AddChannelViewModel.kt

```diff
@@ -0,0 +1,139 @@
+package io.getstream.chat.ui.sample.feature.channel.add
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import io.getstream.chat.android.client.ChatClient
+import io.getstream.chat.android.client.api.models.QuerySort
+import io.getstream.chat.android.client.api.models.QueryUsersRequest
+import io.getstream.chat.android.client.channel.ChannelClient
+import io.getstream.chat.android.client.models.Filters
+import io.getstream.chat.android.client.models.User
+import io.getstream.chat.android.livedata.ChatDomain
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+class AddChannelViewModel : ViewModel() {
+
+    private val currentUserId: String = ChatDomain.instance().currentUser.id
+    private val _state: MutableLiveData<State> = MutableLiveData()
+    private val _paginationState: MutableLiveData<PaginationState> = MutableLiveData()
+    val state: LiveData<State> = _state
+    val paginationState: LiveData<PaginationState> = _paginationState
+    private val usersQueryFilter =
+        Filters.and(Filters.ne("name", ""), Filters.ne("id", currentUserId))
+
+    private var channelClient: ChannelClient? = null
+    private var offset: Int = 0
+
+    init {
+        requestUsers(isRequestingMore = false)
+    }
+
+    fun onEvent(event: Event) {
+        when (event) {
+            Event.ReachedEndOfList -> requestMoreUsers()
+            Event.MessageSent -> createChannel()
+            is Event.MembersChanged -> createDraftChannel(event.members)
+        }
+    }
+
+    private fun requestUsers(isRequestingMore: Boolean) {
+        if (!isRequestingMore) {
+            _state.value = State.Loading
+        }
+        viewModelScope.launch(Dispatchers.IO) {
```

Comment:
We agreed to use `DispatchersProvider` instead of directly using `Dispatchers`

Reply:
`DispatchersProvider` is internal StreamAPI and I don't think we should use it in sample app